projectq/backends/_azure/_azure_quantum_test.py

```diff
@@ -0,0 +1,645 @@
+        verbose=True
+    )
+
+    mapper = BasicMapperEngine()
+    max_qubits = 3
+
+    mapping = {}
+    for i in range(max_qubits):
+        mapping[i] = i
+
+    mapper.current_mapping = mapping
+
+    main_engine = MainEngine(
+        backend=backend,
+        engine_list=[mapper],
+        verbose=True
+    )
+
+    circuit = main_engine.allocate_qureg(3)
+    q0, q1, q2 = circuit
+
+    H | q0  # noqa
+    CX | (q0, q1)  # noqa
+    CX | (q1, q2)  # noqa
+    All(Measure) | circuit
+
+    main_engine.flush()
+
+    assert backend.get_probability('000', circuit) == 0.5
+    assert backend.get_probability('001', circuit) == 0.0
+    assert backend.get_probability('010', circuit) == 0.0
+    assert backend.get_probability('011', circuit) == 0.0
+    assert backend.get_probability('100', circuit) == 0.0
+    assert backend.get_probability('101', circuit) == 0.0
+    assert backend.get_probability('110', circuit) == 0.0
+    assert backend.get_probability('111', circuit) == 0.5
+
+
+@pytest.mark.parametrize(
+    "use_hardware, target_name, provider_id",
+    [
+        (False, 'quantinuum.hqs-lt-s1-apival', 'quantinuum'),
+        (False, 'quantinuum.hqs-lt-s1-sim', 'quantinuum'),
+        (True, 'quantinuum.hqs-lt-s1', 'quantinuum'),
+    ],
+)
+def test_run_quantinuum_get_probability(use_hardware, target_name, provider_id):
+    projectq.backends._azure._azure_quantum.send = mock.MagicMock(
+        return_value={
+            'c': [
+                '000', '000', '000', '111', '111', '000', '000', '111', '000', '111', '000', '000', '111', '000',
+                '111', '111', '111', '000', '111', '111', '000', '111', '000', '111', '111', '111', '000', '111',
+                '000', '111', '111', '111', '111', '111', '000', '111', '111', '111', '111', '111', '111', '111',
+                '000', '000', '111', '000', '111', '111', '000', '000', '000', '000', '000', '111', '000', '111',
+                '000', '111', '111', '111', '111', '111', '111', '111', '000', '111', '000', '111', '111', '111',
+                '111', '000', '000', '000', '000', '111', '111', '000', '111', '111', '000', '000', '111', '000',
+                '111', '000', '111', '000', '111', '111', '000', '111', '000', '111', '111', '111', '000', '111',
+                '111', '000'
+            ]
+        }
+    )
+
+    workspace = mock_workspace()
+
+    backend = AzureQuantumBackend(
+        use_hardware=use_hardware,
+        target_name=target_name,
+        workspace=workspace,
+        verbose=True
+    )
+
+    mapper = BasicMapperEngine()
+    max_qubits = 3
+
+    mapping = {}
+    for i in range(max_qubits):
+        mapping[i] = i
+
+    mapper.current_mapping = mapping
+
+    main_engine = MainEngine(
+        backend=backend,
+        engine_list=[mapper],
+        verbose=True
+    )
+
+    circuit = main_engine.allocate_qureg(3)
+    q0, q1, q2 = circuit
+
+    H | q0  # noqa
+    CX | (q0, q1)  # noqa
+    CX | (q1, q2)  # noqa
```

Comment:
These `noqa` are not be required since we have a general exclusion rule for `useless-statements`

```suggestion
    H | q0
    CX | (q0, q1)
    CX | (q1, q2)
```